server/aap/io/feeding_services/bang.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import lxml.etree
+import requests
+
+from superdesk.errors import AlreadyExistsError
+from superdesk.io.feeding_services.http_base_service import HTTPFeedingServiceBase
+from superdesk.io.iptc import subject_codes
+from superdesk.io.registry import register_feeding_service
+
+
+class BangFeedingService(HTTPFeedingServiceBase):
+    NAME = "Bang"
+
+    label = "Bang Showbiz"
+
+    HTTP_AUTH = False
+
+    MUSIC_ID = "music_url"
+    MOVIES_ID = "movies_url"
+    SHOWBIZ_ID = "showbiz_url"
+
+    session = None
+
+    # configuration fields for the source url's
+    fields = [
+        {
+            "id": MUSIC_ID,
+            "type": "text",
+            "label": "Music URL",
+        },
+        {
+            "id": MOVIES_ID,
+            "type": "text",
+            "label": "Movies URL",
+        },
+        {
+            "id": SHOWBIZ_ID,
+            "type": "text",
+            "label": "Showbiz URL",
+        },
+    ]
+
+    # Map the field/sources entries to appropriate IPTC codes
+    subject_map = {MUSIC_ID: "01011000", MOVIES_ID: "01005001", SHOWBIZ_ID: "01021000"}
+
+    @staticmethod
+    def _config_test(provider=None):
+        return True
+
+    def _update(self, provider, update):
+        if not self.session:
+            self.session = requests.Session()
+
+        parser = self.get_feed_parser(provider)
+
+        items = []
+        for src in self.fields:
+            current_url = provider.get("config").get(src.get("id"))
+            if current_url:
+                provider["current_url"] = current_url
```

Comment:
Or even storing `src["id"]` on the provider, and using that in the URN generation?

Reply:
OK, URN now looks like 

    urn:newsml:music_url:2024-04-23T10:30:00+00:00:BS2BS3062256
    
dumping in the URL looked a bit yuk! From the id I can get which url it was.